Utils/Extensions.Delegate.cs

```diff
@@ -23,8 +23,8 @@ static partial class Extensions {
         /// <param name="method">The method to create the delegate from.</param>
         /// <typeparam name="T">The type of the delegate to create.</typeparam>
         /// <returns>The delegate for this method.</returns>
-        public static Delegate CreateDelegate<T>(this MethodBase method) where T : class
-            => CreateDelegate(method, typeof(T), null);
+        public static T CreateDelegate<T>(this MethodBase method) where T : Delegate
+            => (T) CreateDelegate(method, typeof(T), null);
```

Comment:
This *might* break backwards compatibility with some MMC / MonoMod users, but I'm willing to take the step if you're willing to also apply this to the other generic CreateDelegate signature.

Reply:
I think there may have been a reason that I didn't make the change to the other signature; I'll check.